### DISCUSSION TITLE
Improve skin editor

### DISCUSSION THREAD

**Discussion by nana123126** [ppy/osu#23353] [Ideas]
-Enable the TextElement component to display variables. For example, PP, player name, current time, and so on, players can use {AccPP}, {PP}, {player}, {time}, and so on to express what they want to display.

-The information displayed in BeatmapAttributeText is not accurate when I open some mods.

-Breaking through limitations, allowing parameters of other components such as ColourHitErrorMeter to be directly modified without restrictions.

-Add a component that can display the highest score (insert a digression, can you add a mod in mods, and if cannot exceed a certain score, immediately fail)

-Ultimately, I hope the skin editor can create skins.

(machine translation)

**Comment by ItsShamed** [comment]
I feel like each of these of points could be in their own discussions. Next time please do that. Going to answer anyway..

---

> -Enable the TextElement component to display variables. For example, PP, player name, current time, and so on, players can use {AccPP}, {PP}, {player}, {time}, and so on to express what they want to display.

Some of these were proposed in #22761. Imo the text components should be refactored, adding more interpolation will not work UX-wise with the way it's currently implemented (or how it's suggested to be implemented)

> -The information displayed in BeatmapAttributeText is not accurate when I open some mods.

#21946

> -Breaking through limitations, allowing parameters of other components such as ColourHitErrorMeter to be directly modified without restrictions.

What do you mean by that?

> -Add a component that can display the highest score (insert a digression, can you add a mod in mods, and if cannot exceed a certain score, immediately fail)

If anything, it should be added as an option for the existing components (like the accuracy counter)

> -Ultimately, I hope the skin editor can create skins.

That is the end goal for the most part
